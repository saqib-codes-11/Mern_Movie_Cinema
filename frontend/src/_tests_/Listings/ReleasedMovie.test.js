import ReleasedMovies from "../../Components/Listings/ReleasedMovie";
import { create } from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import React from 'react';

describe(`Unreleased movie testing`, () => {

    let creator;
    let testComponent;

    let movieObj = {
        title: "Fast and Furious 9",
        year: 2021,
        runTime: "143 mins",
        genre: "Action, Adventure, Crime",
        shortPlot: "Cipher enlists the help of Jakob, Dom's younger brother to take revenge on Dom and his team.",
        imageURL: "https://m.media-amazon.com/images/M/MV5BMjI0NmFkYzEtNzU2YS00NTg5LWIwYmMtNmQ1MTU0OGJjOTMxXkEyXkFqcGdeQXVyMjMxOTE0ODA@._V1_SX300.jpg",
        actors: [
            {
                name: "Vin Diesel",
                role: "Dominic Toretto",
                image: "https://www.themoviedb.org/t/p/w600_and_h900_bestv2/ac1dr6oK2ImdcJInKjIZw9sG6Jk.jpg"
            },
            {
                name: "Michelle Rodriguez",
                role: "Letty Ortiz",
                image: "https://www.themoviedb.org/t/p/w600_and_h900_bestv2/soXPKCMmOxT7oyfGsMUHB6YHLcC.jpg"
            },
            {
                name: "Tyrese Gibson",
                role: "Roman Pearce",
                image: "https://www.themoviedb.org/t/p/w600_and_h900_bestv2/jxoy4fbXNKFQtBdK73cLXEz3ufS.jpg"
            }
        ],
        director: "Justin Lin",
        dateTime: [
            {
                day: "Monday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Tuesday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Wednesday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Thursday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Friday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            }
        ],
        released: true,
        classification: "12A"
    }

    it(`should render data is loading`, () => {
        jest.mock('react-router-dom', () => ({
            ...jest.requireActual('react-router-dom'),
            useParams: () => ({
                movie: '60e41cf77a5d6042284b2f33'
            }),
        }));

        const realUseState = React.useState

        jest.spyOn(React, 'useState')
            .mockImplementationOnce(() => realUseState(movieObj))
            .mockImplementationOnce(() => realUseState(null))
            .mockImplementationOnce(() => realUseState(true))

        creator = create(
            <BrowserRouter>
                <ReleasedMovies />
            </BrowserRouter>
        )

        testComponent = creator.root

        const tree = creator.toJSON();
        expect(tree).toMatchSnapshot();
    })

})