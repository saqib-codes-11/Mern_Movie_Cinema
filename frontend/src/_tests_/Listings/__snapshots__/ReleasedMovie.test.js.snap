// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unreleased movie testing should render data is loading 1`] = `
<div
  className="background"
>
  <div
    className="container-fluid"
  >
    <div
      className="pt-5"
    >
      <div
        className=" text-white card"
        style={
          Object {
            "backgroundColor": "#212121",
            "flex": 1,
          }
        }
      >
        <div
          className="no-gutters row"
        >
          <div
            className="col-md-4"
          >
            <img
              alt=" Movie Image"
              height="700px"
              src="https://m.media-amazon.com/images/M/MV5BMjI0NmFkYzEtNzU2YS00NTg5LWIwYmMtNmQ1MTU0OGJjOTMxXkEyXkFqcGdeQXVyMjMxOTE0ODA@.jpg"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="100%"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <div
                className="cardMovieTitle card-title h5"
              >
                Fast and Furious 9
              </div>
              <a
                href="/classifications/2"
                onClick={[Function]}
              >
                <span
                  className="classificationBadge mb-3 badge badge-success"
                  style={
                    Object {
                      "backgroundColor": "#ec7c14",
                    }
                  }
                >
                  Rating: 
                  12A
                </span>
              </a>
              <p
                className="cardMovieText card-text"
              >
                Runtime: 
                143 mins
              </p>
              <p
                className="cardMovieText card-text"
              >
                Directed by 
                Justin Lin
              </p>
              <p
                className="cardMovieText card-text"
              >
                Released: 
                2021
              </p>
              <div
                className="cardMovieText card-subtitle h6"
              >
                Cipher enlists the help of Jakob, Dom's younger brother to take revenge on Dom and his team.
              </div>
              <div
                className="container"
              >
                <div
                  className="row"
                >
                  <div
                    className="py-3 col-lg-2 col-md-3 col-sm-3 col-5"
                  >
                    <div
                      className="actorCard text-white card"
                      style={
                        Object {
                          "backgroundColor": "#912323",
                          "flex": 1,
                        }
                      }
                    >
                      <img
                        class="cardImg"
                        className="card-img"
                        src="https://www.themoviedb.org/t/p/w600_and_h900_bestv2/ac1dr6oK2ImdcJInKjIZw9sG6Jk.jpg"
                      />
                      <div
                        className="card-body"
                      >
                        <div
                          className="card-subtitle h6"
                        >
                          Vin Diesel
                        </div>
                        <p
                          className="card-text"
                        >
                          Dominic Toretto
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    className="py-3 col-lg-2 col-md-3 col-sm-3 col-5"
                  >
                    <div
                      className="actorCard text-white card"
                      style={
                        Object {
                          "backgroundColor": "#912323",
                          "flex": 1,
                        }
                      }
                    >
                      <img
                        class="cardImg"
                        className="card-img"
                        src="https://www.themoviedb.org/t/p/w600_and_h900_bestv2/soXPKCMmOxT7oyfGsMUHB6YHLcC.jpg"
                      />
                      <div
                        className="card-body"
                      >
                        <div
                          className="card-subtitle h6"
                        >
                          Michelle Rodriguez
                        </div>
                        <p
                          className="card-text"
                        >
                          Letty Ortiz
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    className="py-3 col-lg-2 col-md-3 col-sm-3 col-5"
                  >
                    <div
                      className="actorCard text-white card"
                      style={
                        Object {
                          "backgroundColor": "#912323",
                          "flex": 1,
                        }
                      }
                    >
                      <img
                        class="cardImg"
                        className="card-img"
                        src="https://www.themoviedb.org/t/p/w600_and_h900_bestv2/jxoy4fbXNKFQtBdK73cLXEz3ufS.jpg"
                      />
                      <div
                        className="card-body"
                      >
                        <div
                          className="card-subtitle h6"
                        >
                          Tyrese Gibson
                        </div>
                        <p
                          className="card-text"
                        >
                          Roman Pearce
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <h2>
          Book Now!
        </h2>
        <div
          class="container"
        >
          <table
            className="table-hover text-white  table"
            id="tableBorder"
            style={
              Object {
                "backgroundColor": "#A02626",
                "flex": 1,
              }
            }
          >
            <thead>
              <tr>
                <th
                  colSpan="12"
                  scope="colgroup"
                >
                  Showing Times
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  scope="col"
                >
                  Thursday
                   - 
                  15/7
                </th>
                <td>
                  18:00
                </td>
                <td>
                  20:00
                </td>
                <td>
                  22:00
                </td>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Friday
                   - 
                  16/7
                </th>
                <td>
                  18:00
                </td>
                <td>
                  20:00
                </td>
                <td>
                  22:00
                </td>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Monday
                   - 
                  19/7
                </th>
                <td>
                  18:00
                </td>
                <td>
                  20:00
                </td>
                <td>
                  22:00
                </td>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Tuesday
                   - 
                  20/7
                </th>
                <td>
                  18:00
                </td>
                <td>
                  20:00
                </td>
                <td>
                  22:00
                </td>
              </tr>
              <tr>
                <th
                  scope="col"
                >
                  Wednesday
                   - 
                  21/7
                </th>
                <td>
                  18:00
                </td>
                <td>
                  20:00
                </td>
                <td>
                  22:00
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
